spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -339,7 +339,8 @@ private TypeMetadata buildMetadata(Class<?> targetClass) {
 					multiMethods.add(method);
 				}
 			}
-		}, ReflectionUtils.USER_DECLARED_METHODS);
+		}, ReflectionUtils.USER_DECLARED_METHODS
+				.and(meth -> !meth.getDeclaringClass().getName().contains("$MockitoMock$")));
```

Comment:
cherry-picked to 2.4.x breaks compatibility with Spring Framework 5.2, USER_DECLARED_METHODS comes with a `and` method only after 5.3.2

Reply:
Spring Framework 5.2.x is no longer supported as OSS and goes out of commercial support at the end of the year; if you have a commercial support contract with VMware; please raise an issue against spring-amqp via normal procedures there.

https://spring.io/projects/spring-framework#support

Reply:
But the [reference](https://docs.spring.io/spring-amqp/docs/2.4.15/reference/html/#compatibility) said it works with 5.2.x.
And in fact, everything else looks good with 5.2.x

Reply:
That's a bug in the documentation but, in any case, as I said 5.2.x is out of support anyway.

Reply:
I made an exception and made the change https://github.com/spring-projects/spring-amqp/pull/2530/commits/6e18f87258764540ab3f88fbae2a882feb172dfb

The next release is mid-October.

But I still recommend that you move to a supported version of Spring Framework.

Reply:
I made an exception and made the change dcc49ba1b203fa5c79b5a81962e0995725099e64

The next release is mid-October.

But I still recommend that you move to a supported version of Spring Framework.

Reply:
Thank you! I'd like to upgrade, but it's out of my control.